### ISSUE TITLE
cnm2.save fails

### ISSUE THREAD

**Issue by tbenst** [flatironinstitute/CaImAn#427]
- **Tell us about your operating system (Linux/macOS/Windows), Python version (3.x), and working environment (Spyder/Jupyter Notebook/other)**

Linux jupyter

- **Describe the issue that you are experiencing**

Running the demo pipeline notebook on example data works fine, but when using my own tiff file I get this error:
```
ValueError                                Traceback (most recent call last)
<ipython-input-20-859ea2bec651> in <module>
      1 # save the results
      2 hdf5_name = fnames[0][:-4]+"_caiman_results.hdf5"
----> 3 cnm2.save(hdf5_name)
      4 
      5 

~/.conda/envs/CaImAn/lib/python3.6/site-packages/caiman/source_extraction/cnmf/cnmf.py in save(self, filename)
    603 
    604         if '.hdf5' in filename:
--> 605             save_dict_to_hdf5(self.__dict__, filename)
    606         else:
    607             raise Exception("Filename not supported")

~/.conda/envs/CaImAn/lib/python3.6/site-packages/caiman/utils/utils.py in save_dict_to_hdf5(dic, filename)
    357 
    358     with h5py.File(filename, 'w') as h5file:
--> 359         recursively_save_dict_contents_to_group(h5file, '/', dic)
    360 
    361 def load_dict_from_hdf5(filename):

~/.conda/envs/CaImAn/lib/python3.6/site-packages/caiman/utils/utils.py in recursively_save_dict_contents_to_group(h5file, path, dic)
    441             h5file[path + key] = np.array(item)
    442         elif type(item).__name__ in ['CNMFParams', 'Estimates']: # parameter object
--> 443             recursively_save_dict_contents_to_group(h5file, path + key + '/', item.__dict__)
    444         else:
    445             raise ValueError('Cannot save %s type.' % type(item))

~/.conda/envs/CaImAn/lib/python3.6/site-packages/caiman/utils/utils.py in recursively_save_dict_contents_to_group(h5file, path, dic)
    441             h5file[path + key] = np.array(item)
    442         elif type(item).__name__ in ['CNMFParams', 'Estimates']: # parameter object
--> 443             recursively_save_dict_contents_to_group(h5file, path + key + '/', item.__dict__)
    444         else:
    445             raise ValueError('Cannot save %s type.' % type(item))

~/.conda/envs/CaImAn/lib/python3.6/site-packages/caiman/utils/utils.py in recursively_save_dict_contents_to_group(h5file, path, dic)
    424                 h5file[path + key] = item
    425             if not np.array_equal(h5file[path + key].value, item):
--> 426                 raise ValueError('The data representation in the HDF5 file does not match the original dict.')
    427         # save dictionaries
    428         elif isinstance(item, dict):

ValueError: The data representation in the HDF5 file does not match the original dict.
```

My tiffs are of dtype uint16 if that is relevant--seems like the example data is float32?

Edit: [here's the file](https://s3-us-west-2.amazonaws.com/xlfm-dcon/20190107_tyler/f10542_zplane%3D1.tif)

**Comment by agiovann** [comment]
@tbenst could you please paste the full log and not only the error? I need to understand what is the value of "key" and "item" variable. If you can debug that it would help me understand the problem 

Thank you

**Comment by tbenst** [comment]
Hi @agiovann, that is the full output from `cnm2.save(fn)`. I'm not sure what key and value are failing. The relevant line in CaImAn is https://github.com/flatironinstitute/CaImAn/blob/c156373de01a9609d56713b9ed5c754222d2ec8c/caiman/utils/utils.py#L425
Here's the notebook's captured output from during the motion correction + CNMF run itself: https://gist.github.com/tbenst/34426a2c35f71269682f6317c3e6d9aa. Let me know how else I can help. Thank you!

**Comment by agiovann** [comment]
Dear @tbenst ,

I cannot see the failure in the log you sent me for some reason. Did you paste all of it?

**Comment by tbenst** [comment]
@agiovann that's all of it. Logging appears to work properly as https://gist.github.com/tbenst/0d8d3d0ca493c612e1e0f875fa3d6d5b#file-cnmf-debug-log-L5916 to https://gist.github.com/tbenst/0d8d3d0ca493c612e1e0f875fa3d6d5b#file-cnmf-debug-log-L5933 display the INFO level output. Note that only certain keys trigger logging per CaImAn source:

```python
        if key == 'g':
            logging.info(key + ' is an object type')
            item = np.array(list(item))
        if key == 'g_tot':
            item = np.asarray(item, dtype=np.float)
        if key in ['groups', 'idx_tot', 'ind_A', 'Ab_epoch','coordinates','loaded_model', 'optional_outputs','merged_ROIs']:
            logging.info(['groups', 'idx_tot', 'ind_A', 'Ab_epoch', 'coordinates', 'loaded_model', 'optional_outputs', 'merged_ROIs',
                   '** not saved'])
continue
```

**Comment by epnev** [comment]
@tbenst @agiovann This might have been fixed with #444. Try it out when you have a chance.

**Comment by epnev** [comment]
@tbenst Our fix is now pushed to the master branch. I'm closing the issue but feel free to re-open it if you're still encountering issues.
